sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/CPKNTest.groovy

```diff
@@ -0,0 +1,391 @@
+package com.azure.storage.blob
+
+import com.azure.core.http.policy.HttpLogDetailLevel
+import com.azure.core.http.policy.HttpLogOptions
+import com.azure.core.test.TestMode
+import com.azure.core.util.Configuration
+import com.azure.storage.blob.models.BlobContainerEncryptionScope
+import com.azure.storage.blob.models.BlobItem
+import com.azure.storage.blob.models.BlobStorageException
+import com.azure.storage.blob.models.CustomerProvidedKey
```

Comment:
I would prefer to have these test methods sit next to the other tests for the same api (i.e. "append blob create with cpkn" next to the other "append blob create" tests). For better or worse, tests have typically been grouped by api rather than by feature, and I think it'll generally better if we keep to one organizational strategy. If you have a compelling reason to put all the cpkn tests in one file, then I'm open to hearing that.

Reply:
The reason I chose to do this was because this feature is similar to CPK and CPK has its own test class like this